nemo/collections/asr/modules/audio_modules.py

```diff
@@ -0,0 +1,686 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Optional, Tuple
+
+import torch
+
+from nemo.collections.asr.parts.preprocessing.features import make_seq_mask_like
+from nemo.collections.asr.parts.utils.audio_utils import db2mag, wraptopi
+from nemo.core.classes import NeuralModule, typecheck
+from nemo.core.neural_types import AudioSignal, FloatType, LengthsType, NeuralType, SpectrogramType
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+try:
+    import torchaudio
+
+    HAVE_TORCHAUDIO = True
+except ModuleNotFoundError:
+    HAVE_TORCHAUDIO = False
+
+
+__all__ = [
+    'AudioToSpectrogram',
+    'SpectrogramToAudio',
+    'MaskEstimatorRNN',
+    'MaskReferenceChannel',
+    'MaskBasedBeamformer',
+]
+
+
+class AudioToSpectrogram(NeuralModule):
+    """Transform a batch of input multi-channel signals into a batch of
+    STFT-based spectrograms.
+
+    Args:
+        fft_length: length of FFT
+        hop_length: length of hops/shifts of the sliding window
+        power: exponent for magnitude spectrogram. Default `None` will
+               return a complex-valued spectrogram
+    """
+
+    def __init__(self, fft_length: int, hop_length: int, power: Optional[float] = None):
+        if not HAVE_TORCHAUDIO:
+            logging.error('Could not import torchaudio. Some features might not work.')
+
+            raise ModuleNotFoundError(
```

Comment:
Uhh this has a hard requirement of torchaudio ? Cant you use the basic class internally with all the flags set to get back the equivalent without torchaudio.

Note that torchaudio is and will be an optional dependency in NeMo - and this is a major feature (speech enhancement). To keep it behind an external install is painful for the end user. 

There was an external PR which showed how to map the basic one to a torchaudio one for export, you can do the reverse and map this to the basic one to get it to always work without torchaudio.

Ofc if there is no option to do it but to use torchaudio, then ignore this comment.

Reply:
agree with ante, we should start having torchaudio as standard nemo inclusion